openedx/openedx-platform
# fix: log correct time and duration for video subsections

## Description

If a subsection of a video is added to a course using advanced settings, video player shows incorrect time to the user and logs the same time and duration as show to the user, whereas it should log correct time treating the subsection as a standalone video.

## Supporting information

It fixes #35928

## Testing instructions

- Check out master/main
- Add a YouTube video to a course unit and specify a start and end time in advanced settings.
- Preview the video. You will see the video player showing the start and end times you specified rather it should go from zero till the total duration of the clip.
- Inspect the `event` API call using dev tools, you'll notice that the wrong start time and duration are being logged.

Now check out this PR.

- Refresh the page and play the video again.
- You should see the correct time values on the video player.
- Inspect the `event` API call, you should the correct time and duration.
